ui/app/helpers/utils/conversion-util.js

```diff
@@ -150,13 +150,47 @@ const conversionUtil = (value, {
   value: value || '0',
 })
 
+const parseInput = (value) => {
+  const errorMessage = `Invalid BigNumber input: ${value}`
+
+  if (
+    value === null ||
+    value === undefined ||
+    value === '' ||
+    Array.isArray(value)
+  ) {
+    throw new Error(errorMessage)
+  }
+
+  // We don't include 'number' here, because BigNumber will throw if passed
+  // a number primitive it considers unsafe.
+  if (
+    typeof value === 'string' ||
+    value instanceof BigNumber
+  ) {
+    return value
+  }
+
+  let stringValue
+  if (typeof value?.toString === 'function') {
+    stringValue = value.toString()
+  } else {
+    stringValue = String(value)
```

Comment:
I think this is safe to use for both cases. It's `new String()` that produces the odd result of a "string" object that has a type of object